metadata-ingestion/src/datahub/sdk/dataflow.py

```diff
@@ -0,0 +1,303 @@
+from __future__ import annotations
+
+import warnings
+from datetime import datetime
+from typing import Dict, Optional, Type
+
+from typing_extensions import Self
+
+from datahub.cli.cli_utils import first_non_null
+from datahub.emitter.mce_builder import DEFAULT_ENV
+from datahub.errors import (
+    IngestionAttributionWarning,
+)
+from datahub.metadata.schema_classes import (
+    AspectBag,
+    DataFlowInfoClass,
+    EditableDataFlowPropertiesClass,
+)
+from datahub.metadata.urns import DataFlowUrn, Urn
+from datahub.sdk._attribution import is_ingestion_attribution
+from datahub.sdk._shared import (
+    DomainInputType,
+    HasContainer,
+    HasDomain,
+    HasInstitutionalMemory,
+    HasOwnership,
+    HasPlatformInstance,
+    HasSubtype,
+    HasTags,
+    HasTerms,
+    LinksInputType,
+    OwnersInputType,
+    TagsInputType,
+    TermsInputType,
+    make_time_stamp,
+    parse_time_stamp,
+)
+from datahub.sdk.entity import Entity, ExtraAspectsType
+
+
+class DataFlow(
+    HasPlatformInstance,
+    HasSubtype,
+    HasContainer,
+    HasOwnership,
+    HasInstitutionalMemory,
+    HasTags,
+    HasTerms,
+    HasDomain,
+    Entity,
+):
+    """Represents a dataflow in DataHub.
+    A dataflow represents a collection of data, such as a table, view, or file.
+    This class provides methods for managing dataflow metadata including schema,
+    lineage, and various aspects like ownership, tags, and terms.
+    """
+
+    __slots__ = ()
+
+    @classmethod
+    def get_urn_type(cls) -> Type[DataFlowUrn]:
+        """Get the URN type for dataflows.
+        Returns:
+            The DataflowUrn class.
+        """
+        return DataFlowUrn
+
+    def __init__(
+        self,
+        *,
+        # Identity.
+        id: str,
```

Comment:
how do we make it more clear what the user is supposed to provide for the `id` field?

Reply:
switched to name and added display_name